diff --git a/handlers.go b/handlers.go
index 72bf6c4..c1264fd 100644
--- a/handlers.go
+++ b/handlers.go
@@ -460,6 +460,28 @@ func handleLive(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func handleStats(w http.ResponseWriter, r *http.Request) {
+	type Response struct {
+		StreamStart int64 `json:"streamStart"`
+		StreamLive  bool  `json:"streamLive"`
+	}
+	res := &Response{}
+
+	stats.mutex.Lock()
+	res.StreamStart = stats.streamStart.Unix()
+	res.StreamLive = stats.streamLive
+	stats.mutex.Unlock()
+	
+	resJson, err := json.Marshal(res)
+	if err != nil {
+		common.LogErrorf("Error generating stats JSON: %s\n", err)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(200)
+	w.Write(resJson)
+}
+
 func handleDefault(w http.ResponseWriter, r *http.Request) {
 	if r.URL.Path != "/" {
 		// not really an error for the server, but for the client.
diff --git a/main.go b/main.go
index 662f2a6..0e9623c 100644
--- a/main.go
+++ b/main.go
@@ -151,6 +151,7 @@ func run(args args) {
 	router.HandleFunc("/help", wrapAuth(handleHelpTemplate))
 	router.HandleFunc("/emotes", wrapAuth(handleEmoteTemplate))
 
+	router.HandleFunc("/stats", wrapAuth(handleStats))
 	router.HandleFunc("/live", wrapAuth(handleLive))
 	router.HandleFunc("/", wrapAuth(handleDefault))
 
